### ISSUE TITLE
[BUG] op.to_type() throws error when creating a private entry

### ISSUE THREAD

**Issue by mattyg** [holochain/holochain#1861]
**Describe the bug**
When creating a private entry, I always get the following validation error on my local conductor. It occurs when I call `to_type()` in my validate function:

```
Guest("Entry type is a capability and should be private but there is an entry present")
```

**Expected behavior**
Calling `to_type()` does not raise validation error.

**System information:**
 - OS: ubuntu 22.04
 - Holochain and HDK Version: 0.1.0

**Steps to reproduce**
Minimum reproduction: https://github.com/mattyg/privateentry-bug-reproduction

**Additional context**
I believe the error I'm getting is from here: https://github.com/holochain/holochain/blob/11fd977aa6c4e97a6a9e7f261db06060b80d9d82/crates/hdi/src/op.rs#L589

Because the validation is run on the author's cell, the RecordEntryRef is Present. I'm not sure if it's supposed to be and we just need another match arm, or there is a different underlying issue.

**Comment by abe-njama** [comment]
@mattyg Thanks you for raising this. We're getting this looked into and @maackle is running point on debugging and working on a fix for this.

**Comment by maackle** [comment]
Thanks for this great reproduction. It's as you say. `Op::to_type` assumes that the StoreEntry op will never contain a private entry, when in fact it does (only for the author, to give them a chance to validate their own private entry). So we need to make all code paths consistent with this reality.

**Comment by maackle** [comment]
The fix is done via #1910, but it is a slightly breaking change to the `OpType` structure, so I imagine it will be included in the 0.2 release and not backported to 0.1.x unless needed.

**Comment by maackle** [comment]
Should be fixed.
